source/includes/fundamentals/code-snippets/CRUD/distinctValues.go

```diff
@@ -57,13 +57,12 @@ func main() {
 	// Retrieves and prints distinct values of the "department" field in
 	// documents that match the filter
 	// begin distinct
-	results, err := coll.Distinct(context.TODO(), "department", bson.D{{"enrollment", bson.D{{"$lt", 50}}}})
-	if err != nil {
-		panic(err)
-	}
+	filter := bson.D{{"enrollment", bson.D{{"$lt", 50}}}}
+	res := coll.Distinct(context.TODO(), "department", filter)
 
-	for _, result := range results {
-		fmt.Println(result)
-	}
+	var arr []string
+	res.Decode(&arr)
```

Comment:
We still need to check the error here. Also, consider using the more common pattern of chaining the `Distinct` call.

For example:
```go
var arr []string
err := coll.Distinct(context.TODO(), "department", filter).Decode(&arr)
if err != nil {
	panic(err)
}
```